src/components/ui/action-icon.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from "react"
+import { cn } from "@/lib/utils"
+import { Icon, IconType } from "./icon"
+
+
+export enum ActionType {
+  ApiCall = "API_CALL",
+  ApiCallGraphql = "API_CALL_GRAPHQL",
+  Calculation = "CALCULATION",
+  Checklist = "CHECKLIST",
+  ClinicalNote = "CLINICAL_NOTE",
+  Form = "FORM",
+  Message = "MESSAGE",
+  Plugin = "PLUGIN",
+  PushToEmr = "PUSH_TO_EMR",
+}
+
+export interface ActionIconProps {
+    actionType: ActionType
+    icon_url?: string
+}
+
+const ActionIcon = React.forwardRef<HTMLDivElement, ActionIconProps>(
+  ({ ...props }, _ref) => {
+    const TYPES_ICONS_MAP: Record<
+      ActionType,
+      { icon: IconType }
+    > = {
+      [ActionType.ApiCall]: {
+        icon: "RiTerminalBoxFill"
+      },
+      [ActionType.ApiCallGraphql]: {
+        icon: "RiTerminalWindowFill"
+      },
+      [ActionType.Calculation]: {
+        icon: "RiCalculatorFill"
+      },
+      [ActionType.Checklist]: {
+        icon: "RiCheckboxMultipleFill"
+      },
+      [ActionType.ClinicalNote]: {
+        icon: "RiHealthBookFill"
+      },
+      [ActionType.Form]: {
+        icon: "RiSurveyFill"
+      },
+      [ActionType.Message]: {
+        icon: "RiMessage3Fill"
+      },
+      [ActionType.Plugin]: {
+        icon: "RiUploadCloudFill"
+      },
+      [ActionType.PushToEmr]: {
+        icon: "RiUploadCloudFill"
+      },
```

Comment:
If we have only icon props we can simplify object to (directly assign icon value):
```js
[ActionType.ApiCall]: "RiTerminalBoxFill",
[ActionType.ApiCallGraphql]: "RiTerminalWindowFill"
....
```